src/mame/drivers/ms0515.cpp

```diff
@@ -115,9 +211,128 @@ WRITE16_MEMBER(ms0515_state::ms0515_bank_w)
 	}
 }
 
-WRITE8_MEMBER(ms0515_state::ms0515_sys_w)
+WRITE16_MEMBER(ms0515_state::ms0515_vram_w)
+{
+	m_video_ram[offset*2]   = data;
+	m_video_ram[offset*2+1] = data >> 8;
+}
+
+READ16_MEMBER(ms0515_state::ms0515_halt_r)
+{
+	DBG_LOG(2,"Halt",("== %06o\n", m_haltreg));
+
+	return m_haltreg;
+}
+
+WRITE16_MEMBER(ms0515_state::ms0515_halt_w)
 {
-	m_sysreg = data;
+	DBG_LOG(1,"Halt",("<- %06o\n", data));
+
+	COMBINE_DATA(&m_haltreg);
+}
+
+/*
+ * b7 -- ROM bank
+ * b6 -- cassette data out
+ * b5 -- LED VD16
+ * b4 -- LED VD9
+ * b3 -- floppy side select (?? 1 -- top)
+ * b2 -- floppy motor (0 -- on)
+ * b1-0 -- floppy drive select
+ *
+ * DZ0 = drive 0 side 0 (bottom)
+ * DZ1 = drive 1 side 0 (bottom)
+ * DZ2 = drive 0 side 1 (top)
+ * DZ3 = drive 1 side 1 (top)
+ *
+ * MZ1 = drive 1 side 0-1
+ */
+WRITE8_MEMBER(ms0515_state::ms0515_porta_w)
+{
+	DBG_LOG(2,"Sysreg",("A <- %02x\n", data));
+
+	output().set_value("led16", BIT(data, 5));
+	output().set_value("led9", BIT(data, 4));
+
+	switch (data & 3)
+	{
+	case 0:
+		m_floppy = m_floppy0;
+		break;
+
+	case 1:
+		m_floppy = m_floppy1;
+		break;
+
+	default:
+		m_floppy = nullptr;
+		break;
+	}
+
+	if (m_floppy) {
```

Comment:
Once again, you've got a grating mix of Allman and K&R - please make it consistent.